LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2Effect_Sustain_LW.uc

```diff
@@ -0,0 +1,48 @@
+class X2Effect_Sustain_LW extends X2Effect_Sustain;
+
+function bool PreDeathCheck(XComGameState NewGameState, XComGameState_Unit UnitState, XComGameState_Effect EffectState)
+{
+	local X2EventManager EventMan;
+	local UnitValue SustainValue;
+	local int Index, PercentChance, RandRoll;
+    local UnitValue OverKillDamage;
+	if( !UnitState.IsAbleToAct(true) )
+	{
+		// Stunned units may not go into Sustain
+		return false;
+	}
+
+	if (UnitState.GetUnitValue(default.SustainUsed, SustainValue))
+	{
+		if (SustainValue.fValue > 0)
+			return false;
+    }
+        
+    UnitState.GetUnitValue('OverKillDamage', OverKillDamage);
+
+	Index = default.SUSTAINTRIGGERUNITCHECK_ARRAY.Find('UnitType', UnitState.GetMyTemplateName());
+
+	// If the Unit Type is not in the array, then it always triggers sustain
+	if (Index != INDEX_NONE)
+	{
+		PercentChance = 100 - (default.SUSTAINTRIGGERUNITCHECK_ARRAY[Index].PercentChance * OverKillDamage.fValue);
```

Comment:
I'm not sure about changing the nature/meaning of this config array. It will be confusing for folks that want to change it and are used to the vanilla behaviour. It may also adversely affect mods that use that config array for other purposes.

It's more work, but I think it would be better to introduce a new array for this information that makes it clear it's LW specific.